### ISSUE TITLE
make test fails on Windows

### ISSUE THREAD

**Issue by millerbryan** [neilb/IO-Tee#3]
Please add a disclaimer to the IO::Tee page noting that Windows is not supported.   I see it failed CPAN tests on 32-bit Windows and I am experiencing failures on 64-bit Windows under Strawberry Perl.

cpan> install IO::Tee
Running install for module 'IO::Tee'
Fetching with LWP:
http://cpan.strawberryperl.com/authors/id/N/NE/NEILB/IO-Tee-0.65.tar.gz
Fetching with LWP:
http://cpan.strawberryperl.com/authors/id/N/NE/NEILB/CHECKSUMS
Checksum for C:\STRAWB~1\cpan\sources\authors\id\N\NE\NEILB\IO-Tee-0.65.tar.gz ok
Configuring N/NE/NEILB/IO-Tee-0.65.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for IO::Tee
Writing MYMETA.yml and MYMETA.json
  NEILB/IO-Tee-0.65.tar.gz
  C:\Strawberry\perl\bin\perl.exe Makefile.PL -- OK
Running make for N/NE/NEILB/IO-Tee-0.65.tar.gz
cp lib/IO/Tee.pm blib\lib\IO\Tee.pm
  NEILB/IO-Tee-0.65.tar.gz
  C:\STRAWB~1\c\bin\gmake.exe -- OK
Running make test
"C:\Strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-
e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.tt/original.t .. Failed 1/27 subtests

Test Summary Report
-------------------
t/original.t (Wstat: 0 Tests: 27 Failed: 1)
  Failed test:  26
Files=1, Tests=27,  0 wallclock secs ( 0.05 usr +  0.02 sys =  0.06 CPU)
Result: FAIL
Failed 1/1 test programs. 1/27 subtests failed.
gmake: *** [Makefile:861: test_dynamic] Error 255
  NEILB/IO-Tee-0.65.tar.gz
  C:\STRAWB~1\c\bin\gmake.exe test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
  reports NEILB/IO-Tee-0.65.tar.gz
Stopping: 'install' failed for 'IO::Tee'.
Failed during this command:
 NEILB/IO-Tee-0.65.tar.gz                     : make_test NO

**Comment by EstefanoPPT** [comment]
Ditto here!  Using cpan to install IO::Tee failed on Windows 10 x64, using Strawberry Perl 5.26.1.1-64bit.

```
> cpan IO::Tee
Loading internal null logger. Install Log::Log4perl for logging messages
CPAN: CPAN::SQLite loaded ok (v0.211)
Database was generated on Mon, 12 Mar 2018 13:43:09 GMT

Running install for module 'IO::Tee'
CPAN: Digest::SHA loaded ok (v5.97)
CPAN: Compress::Zlib loaded ok (v2.074)
Checksum for C:\STRAWB~1\cpan\sources\authors\id\N\NE\NEILB\IO-Tee-0.65.tar.gz ok
CPAN: Archive::Tar loaded ok (v2.26)
CPAN: YAML::XS loaded ok (v0.66)
CPAN: CPAN::Meta::Requirements loaded ok (v2.140)
CPAN: Parse::CPAN::Meta loaded ok (v2.150010)
CPAN: CPAN::Meta loaded ok (v2.150010)
CPAN: Module::CoreList loaded ok (v5.20180220)
Configuring N/NE/NEILB/IO-Tee-0.65.tar.gz with Makefile.PL
Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for IO::Tee
Writing MYMETA.yml and MYMETA.json
  NEILB/IO-Tee-0.65.tar.gz
  C:\Strawberry\perl\bin\perl.exe Makefile.PL -- OK
Running make for N/NE/NEILB/IO-Tee-0.65.tar.gz
cp lib/IO/Tee.pm blib\lib\IO\Tee.pm
  NEILB/IO-Tee-0.65.tar.gz
  C:\STRAWB~1\c\bin\gmake.exe -- OK
Running make test
"C:\Strawberry\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
t/original.t .. Failed 1/27 subtests

Test Summary Report
-------------------
t/original.t (Wstat: 0 Tests: 27 Failed: 1)
  Failed test:  26
Files=1, Tests=27,  0 wallclock secs ( 0.03 usr +  0.00 sys =  0.03 CPU)
Result: FAIL
Failed 1/1 test programs. 1/27 subtests failed.
gmake: *** [Makefile:861: test_dynamic] Error 255
  NEILB/IO-Tee-0.65.tar.gz
  C:\STRAWB~1\c\bin\gmake.exe test -- NOT OK
//hint// to see the cpan-testers results for installing this module, try:
  reports NEILB/IO-Tee-0.65.tar.gz
Stopping: 'install' failed for 'IO::Tee'.
```

**Comment by hexcoder-** [comment]
I analysed the issue with the failing test. Under windows sysread() does not treat line endings like read() does. After adjusting this, the tests all pass.

[original.zip](https://github.com/neilb/IO-Tee/files/1883960/original.zip)

**Comment by millerbryan** [comment]
@hexcoder- Can you post a patch or comment on what files were impacted?  I can submit a pull request if don't plan to as I would really like to see this working again.

Thanks!

**Comment by hexcoder-** [comment]
[original.t.patch.zip](https://github.com/neilb/IO-Tee/files/1893155/original.t.patch.zip)

Hi Bryan,

the patch is attached.
IMHO the test case (26) in file ‘t/original.t’ is not done portably.

For test 26 I added an extra line to change Windows/DOS-like line endings to „\n“ in the variable $contents, which was filled by sysread().
Since sysread() does no translation from raw line endings (CRLF) to logical endings (“\n”) like read() does,
there are differences on operating systems with different native line endings.

The change could be made conditional for DOS-like operating systems, if needed.

I hope you agree.
It would be nice, if you could do the pull request then.

Greetings Heiko (hexcoder)

Von: millerbryan [mailto:notifications@github.com]
Gesendet: Samstag, 7. April 2018 17:40
An: neilb/IO-Tee
Cc: hexcoder; Mention
Betreff: Re: [neilb/IO-Tee] make test fails on Windows (#3)


@hexcoder-<https://github.com/hexcoder-> Can you post a patch or comment on what files were impacted? I can submit a pull request if don't plan to as I would really like to see this working again.

Thanks!

**Comment by hexcoder-** [comment]
First I replied by email with the patch attachment. Since I don't see it here :-(((, I attached the patch again to comment above (as original.t.patch.zip).
Please confirm, if you have everything you need for the pull request!

Thanks, hexcoder

**Comment by neilb** [comment]
Hi -- I'll check this later -- thanks!

**Comment by millerbryan** [comment]
I believe I have everything I need.  It may be a few days until I can work on this as I am travelling.

**Comment by harriott** [comment]
While awaiting implementation of @hexcoder's patch, here is how it worked for me:
1. `Windows 10 x64, version 1709`
2. `strawberry-perl-5.26.1.1-64bit.msi`
3. `cpanm IO::Tee` failed test number 26 out of 27, but left behind `C:\Users\jnbh\.cpanm\work\1523991036.9660\IO-Tee-0.65`, so in there:
    1. applied @hexcoder's two-line patch to `t\original.t`
    2. (following clues in `README`: `gmake`, `gmake test`, `gmake install`

**Comment by hexcoder-** [comment]
Seems like this one got forgotten. Come on, it shouldn't be rocket science.

**Comment by beppe9000** [comment]
bump. i got this error. is it ok to `--force` the install ?

**Comment by millerbryan** [comment]
There is a two-line patch above as provided by hexcoder.  Unfortunately, a year has gone by and the patch has not been applied.

**Comment by vladimir-shmidt** [comment]
Unfortunately, two years have gone by and the patch has not been applied.

https://github.com/gitpan/IO-Tee/pull/1

**Comment by neilb** [comment]
Fixed in 0.66, now on CPAN.
